app/javascript/gobierto_data/webapp/components/Dataset.vue

```diff
@@ -876,29 +899,51 @@ export default {
     isSavingPromptVisibleHandler(value) {
       this.isSavingPromptVisible = value
     },
-    queryIsNotMine() {
+    queryOrVizIsNotMine() {
       const userId = Number(getUserId());
       const {
         params: { queryId },
         name: nameComponent
       } = this.$route;
 
-      let items = this.publicQueries;
+      //Find which query is loaded
+      if (userId !== 0 && nameComponent === 'Query') {
 
-      //Find which query is loaded, get the userId and queryId
-      const { id: oldQueryId, attributes: { user_id: checkUserId } = {} } = items.find(({ id }) => id === queryId) || {}
+        const items = !this.showPrivate ? this.publicQueries : this.privateQueries
+        const { attributes: { user_id: checkUserId } = {} } = items.find(({ id }) => id === queryId) || {}
 
-      /*Check:
-      - If user is logged
-      - If the user who loaded the query is the same user who created the query
-      - If the component is query
-      - The user can fork a query from another user, so we need to check if the ID's are equal.*/
-      if (this.isUserLogged && userId !== checkUserId && nameComponent === 'Query' && oldQueryId === queryId) {
-        this.enabledForkButton = true
-        this.isForkPromptVisible = true
-      } else {
-        this.disabledForkButton()
-        this.enabledForkPrompt()
+        //Check if the user who loaded the query is the same user who created the query
+        if (userId !== checkUserId) {
+          this.showPrivatePublicIcon = false
+          this.enabledForkButton = true
+          this.isForkPromptVisible = true
+        } else {
+          this.showPrivatePublicIcon = true
+          this.disabledForkButton()
+        }
+      } else if (userId !== 0 && nameComponent === 'Visualization') {
+
+        this.showLabelEdit = true
+
+        const objectViz = this.privateVisualizations.find(({ id }) => id === queryId) || {}
```

Comment:
This whole file has become massive. Perhaps, you should think about splitting the logic in Queries and Visualizations, create both mixins where you set all the stuff related (events, attributes) and maybe you'll avoid the conditional clauses all around.

Think about it, it would deserve a PR to make it happen.